core/src/com/unciv/models/metadata/GameSettings.kt

```diff
@@ -68,4 +74,64 @@ class GameSettings {
         if (tutorialTasksCompleted.add(tutorialTask))
             save()
     }
+
+    fun updateLocaleFromLanguage() {
+        val bannedCharacters = listOf(' ', '_', '-', '(', ')') // Things not to have in enum names
+        val languageName = language.filterNot { it in bannedCharacters }
+        val code = try {
+            LocaleCode.valueOf(languageName)
+        } catch (e: IllegalArgumentException) {
+            LocaleCode.English
+        }
+        locale = Locale(code.language, code.country)
+    }
+
+    fun getCurrentLocale(): Locale {
+        if (locale == null)
+            updateLocaleFromLanguage()
+        return locale!!
+    }
+}
+
+enum class LocaleCode(var language: String, var country: String) {
+    Arabic("ar", "IQ"),
+    BrazilianPortuguese("pt", "BR"),
+    Bulgarian("bg", "BG"),
+    Catalan("ca", "ES"),
+    Croatian("hr", "HR"),
+    Czech("cs", "CZ"),
+    Danish("da", "DK"),
+    Dutch("nl", "NL"),
+    English("en", "US"),
+    Estonian("et", "EE"),
+    Finnish("fi", "FI"),
+    French("fr", "FR"),
+    German("de", "DE"),
+    Greek("el", "GR"),
+    Hindi("hi", "IN"),
+    Hungarian("hu", "HU"),
+    Indonesian("in", "ID"),
+    Italian("it", "IT"),
+    Japanese("ja", "JP"),
+    Korean("ko", "KR"),
+    Latvian("lv", "LV"),
+    Lithuanian("lt", "LT"),
+    Malay("ms", "MY"),
+    Norwegian("no", "NO"),
+    NorwegianNynorsk("nn", "NO"),
+    // Sorry Persian is not in the supported locales
```

Comment:
Maybe called Iranian?

Reply:
[Farsi](https://www.localeplanet.com/icu/fa-IR/index.html) - wonder if Collator.getInstance accepts "fa"

Reply:
OK, fa_IR doesn't crash on desktop or Oreo. Too lazy to fire up a Lollipop or thereabouts AVM. Of course, how it actually sorts - who knows. But moving the Locale(code.language, code.country) call inside the try should cover any stone age java having a shorter locale list. So a best-effort implementation for Persian is possible. As Persian sits at 20% I guess we have no tester to ask.

Reply:
Literally nothing seems to crash the Locale constructor, it seems to fall back to some default value. English or system default? Iranian or Farsi is not in the list of supported languages over at Oracle, falling back to system locale is probably best option.